### PR TITLE
zib-MedicationUse

### PR DESCRIPTION
**Pull request by pieter-edelman-nictiz**

### CODE REVIEW

File: resources/nl-core/nl-core-MedicationUse.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      <path value="MedicationStatement.extension" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="url" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="MedicationStatement.extension:medicationTreatment">
```
Comment (pieter-edelman-nictiz) [inline resources/nl-core/nl-core-MedicationUse.xml:41]: Could you point to where this extension is defined?

File: resources/zib/zib-MedicationUse.xml [hunk 2]
```diff
@@ -0,0 +1,275 @@
+<?xml version="1.0" encoding="utf-8"?>
+<StructureDefinition xmlns="http://hl7.org/fhir">
+  <id value="zib-MedicationUse" />
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: You are completely right, overlooked it!

File: resources/zib/ext-MedicationUse.Prescriber.xml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    <element id="Extension.value[x]">
+      <path value="Extension.value[x]" />
+      <short value="Prescriber" />
+      <definition value="The health professional that entered the medication agreement with the patient." />
+      <alias value="Voorschrijver" />
+      <type>
+        <code value="Reference" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/pattern-ZibHealthProfessionalReference" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-HealthProfessional-PractitionerRole" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-HealthProfessional-Practitioner" />
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/ext-MedicationUse.Prescriber.xml]: I think I forgot to remove the reference. Just did it.

File: resources/zib/zib-MedicationUse.xml [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    </element>
+    <element id="MedicationStatement.extension:prescriber">
+      <path value="MedicationStatement.extension" />
+      <sliceName value="prescriber" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-MedicationUse.Prescriber" />
+      </type>
+    </element>
+    <element id="MedicationStatement.extension:asAgreedIndicator">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Fixed, thnx.

File: resources/zib/zib-MedicationUse.xml [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+      <path value="MedicationStatement.extension" />
+      <slicing>
+        <discriminator>
+          <type value="value" />
+          <path value="url" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="MedicationStatement.extension:instructionsForUseDescription">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: No, it should cover all occurrences of `.dosage`. The information would be duplicated if it is inside the .dosage. Tried to explain this in the InstructionForUse root element.

File: resources/zib/zib-MedicationUse.xml [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+    <element id="MedicationStatement.modifierExtension:stopType">
+      <path value="MedicationStatement.modifierExtension" />
+      <sliceName value="stopType" />
+      <max value="1" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-StopType" />
+      </type>
+    </element>
+    <element id="MedicationStatement.modifierExtension:repeatPeriodCyclicalSchedule">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Same as above.

File: resources/zib/zib-MedicationUse.xml [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-InstructionsForUse.Description" />
+      </type>
+    </element>
+    <element id="MedicationStatement.extension:periodOfUseDuration">
+      <path value="MedicationStatement.extension" />
+      <sliceName value="periodOfUseDuration" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-PeriodOfUse-Duration" />
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: It lives in its own branch. We will need to discuss the best approach for this.

File: resources/nl-core/nl-core-MedicationUse.xml (hunk 1 above)
Comment (ArdonToonstra) [reply resources/nl-core/nl-core-MedicationUse.xml]: In it's own branch :)

File: resources/zib/zib-MedicationUse.xml [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+      <short value="MedicationUseDateTime" />
+      <definition value="Date on which this use is entered." />
+      <alias value="MedicatieGebruikDatumTijd" />
+      <mapping>
+        <identity value="zib-medicationuse2-v1.1.1-2020EN" />
+        <map value="NL-CM:9.11.22398" />
+        <comment value="MedicationUseDateTime" />
+      </mapping>
+    </element>
+    <element id="MedicationStatement.reasonCode">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Done!

File: resources/zib/zib-MedicationUse.xml [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+    <uri value="https://zibs.nl/wiki/TimeInterval-v1.0(2020EN)" />
+    <name value="zib TimeInterval-v1.0(2020EN)" />
+  </mapping>
+  <kind value="resource" />
+  <abstract value="true" />
+  <type value="MedicationStatement" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/MedicationStatement" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="MedicationStatement">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Done. with a space...

File: resources/zib/zib-MedicationUse.xml [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+    <element id="MedicationStatement.modifierExtension:useIndicator">
+      <path value="MedicationStatement.modifierExtension" />
+      <sliceName value="useIndicator" />
+      <max value="1" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-MedicationUse.UseIndicator" />
+      </type>
+    </element>
+    <element id="MedicationStatement.modifierExtension:stopType">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Entered in errror indeed. Fixed.

File: resources/zib/zib-MedicationUse.xml [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+          <type value="value" />
+          <path value="url" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="MedicationStatement.modifierExtension:useIndicator">
+      <path value="MedicationStatement.modifierExtension" />
+      <sliceName value="useIndicator" />
+      <max value="1" />
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Discussed this with Arianne en Marijke. They have written down all the use cases here: https://informatiestandaarden.nictiz.nl/wiki/mp:V9.1.0_Ontwerp_medicatieproces#Medicatiegebruik:_gebruikindicator.2C_volgens_afspraak_indicator.2C_stoptype.2C_gebruiksperiode_en_doseerinstructie

I think the mapping can work... What to do with the status values completed, entered-in-error and intended? Would hat imply that UseIndicator = false too?

File: resources/zib/zib-MedicationUse.xml [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+    <element id="MedicationStatement.modifierExtension:repeatPeriodCyclicalSchedule">
+      <path value="MedicationStatement.modifierExtension" />
+      <sliceName value="repeatPeriodCyclicalSchedule" />
+      <max value="1" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://nictiz.nl/fhir/StructureDefinition/ext-InstructionsForUse.RepeatPeriodCyclicalSchedule" />
+      </type>
+    </element>
+    <element id="MedicationStatement.status">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Done!

File: resources/zib/zib-MedicationUse.xml [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+        <targetProfile value="http://hl7.org/fhir/StructureDefinition/Medication" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-PharmaceuticalProduct" />
+      </type>
+      <mapping>
+        <identity value="zib-medicationuse2-v1.1.1-2020EN" />
+        <map value="NL-CM:9.11.21338" />
+        <comment value="MedicationUse" />
+      </mapping>
+    </element>
+    <element id="MedicationStatement.effective[x]">
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-MedicationUse.xml]: Is handled through branch / PR named zib-TimeInterval

File: resources/zib/zib-MedicationUse.xml (hunk 11 above)
[the remark this answers is not in the document]
Comment (pieter-edelman-nictiz) [reply resources/zib/zib-MedicationUse.xml]: I guess so ...

[the remark this answers is not in the document]
Comment (pieter-edelman-nictiz) [reply resources/zib/zib-MedicationUse.xml]: I think the useIndicator extension can be removed as well.

**Comment by ArdonToonstra** [comment]
Will be merged into branch MedicationProfiles

**Pull request closed without merging**